mqt/qfr/qiskit/QuantumCircuit.hpp

```diff
@@ -302,25 +319,20 @@ namespace qc::qiskit {
             qubits.pop_back();
             auto target0 = qubits.back().qubit;
             qubits.pop_back();
-            qc::SymbolOrNumber theta  = 0.;
-            qc::SymbolOrNumber phi    = 0.;
-            qc::SymbolOrNumber lambda = 0.;
-            if (params.size() == 1) {
-                lambda = parseParam(params[0]);
-            } else if (params.size() == 2) {
-                phi    = parseParam(params[0]);
-                lambda = parseParam(params[1]);
-            } else if (params.size() == 3) {
-                theta  = parseParam(params[0]);
-                phi    = parseParam(params[1]);
-                lambda = parseParam(params[2]);
+            std::vector<qc::SymbolOrNumber> parameters{};
+            for (std::size_t i = 0; i < params.size(); ++i) { // NOLINT(modernize-loop-convert)
+                parameters.emplace_back(parseParam(params[i]));
             }
```

Comment:
Doesn't this work as `for-each` loop?

Reply:
I didn't get it to work at first because range-based for loops over pybind types are weird. 
Thanks to you poking me, I got it to work in https://github.com/cda-tum/qfr/pull/280/commits/4b36b199fe857eb6063b0debba426e71db3480fb.